### PR TITLE
pcm_converter: Fix architectures not supporting XT_ROUND_S

### PR DESCRIPTION
**Pull request by hongshui3000**
XT_ROUND_S is available only when you include option of
"Xtensa's single precision FPU option (scalar)".
IF current DSP includes optional vector FPU (single precision).
FIROUND.S (with appropriate RoundMode state set) followed by
TRUNC.S can be used to emulate Xtensa scalar FPU’s ROUND.S, CEIL.S and
FLOOR.S instructions."FIROUND.S", does round, but no scaling.Result
still in float format.Use TRUNC.S to scale it and truncate as necessary.
Fix :https://github.com/thesofproject/sof/issues/4042

Signed-off-by: Yang XiaoHua <yangxiaohuamail@gmail.com>

### CODE REVIEW

**Review by paulstelian97 — COMMENTED** [review]
Almost perfect, however in the commit message I can see FIRINT.S and FIROUND.S for some reason, and only FIROUND.S is referenced from the code itself. This feels inconsistent. Typo or something else that needs explanation?

**Comment by hongshui3000** [comment]
> Almost perfect, however in the commit message I can see FIRINT.S and FIROUND.S for some reason, and only FIROUND.S is referenced from the code itself. This feels inconsistent. Typo or something else that needs explanation?

It should be both can achieve the purpose, but my code chooses the latter. But I am not a dsp expert, I hope someone in this area can help me review

**Comment by hongshui3000** [comment]
> Almost perfect, however in the commit message I can see FIRINT.S and FIROUND.S for some reason, and only FIROUND.S is referenced from the code itself. This feels inconsistent. Typo or something else that needs explanation?

Corrected

**Comment by hongshui3000** [comment]
I need a DSP expert to help me check the code

**Comment by paulstelian97** [comment]
Do also take care of the Checkpatch issues.

File: src/audio/pcm_converter/pcm_converter_hifi3.c
Comment (dbaluta) [inline src/audio/pcm_converter/pcm_converter_hifi3.c:679]: Firstly please trim the commit subject to around 100 chars. 

E.g: `pcm_converter: Fix architectures not supporting XT_ROUND_S`

Please also, fix checkpatch.pl issues. Also, please be patient.

**Comment by lgirdwood** [comment]
> I need a DSP expert to help me check the code

@singalsu can you check.

File: src/audio/pcm_converter/pcm_converter_hifi3.c
Comment (lgirdwood) [inline src/audio/pcm_converter/pcm_converter_hifi3.c:785]: Indentation looks wrong here and in the other places. Tabs are used with size 8.

**Review by lgirdwood — COMMENTED** [review]
LGTM except for indentation, but wont merge until the HiFi experts can approve.

File: src/audio/pcm_converter/pcm_converter_hifi3.c
Comment (hongshui3000) [reply src/audio/pcm_converter/pcm_converter_hifi3.c:785]: Some obvious problems, I will update later

File: src/audio/pcm_converter/pcm_converter_hifi3.c
Comment (hongshui3000) [reply src/audio/pcm_converter/pcm_converter_hifi3.c:679]: Some obvious problems, I will update later

**Review by singalsu — COMMENTED** [review]
What if used xtensa version supports all of these three instructions? If XT_ROUND_S() exists it should be the preference since it is single instruction. If it doesn't exist, check for XT_FIROUND(). Otherwise error. Due to this, an inline function for all occurrences would be good.

I did check the platforms we are working with do not have XT_FIROUND_S() so there's no this issue but in future we could have.

Can you also help me find document for XT_FIROUND_S()? I can't seem to find even from latest RI-2021.6 tools.

**Comment by hongshui3000** [comment]
> What if used xtensa version supports all of these three instructions? If XT_ROUND_S() exists it should be the preference since it is single instruction. If it doesn't exist, check for XT_FIROUND(). Otherwise error. Due to this, an inline function for all occurrences would be good.
> 
> I did check the platforms we are working with do not have XT_FIROUND_S() so there's no this issue but in future we could have.
> 
> Can you also help me find document for XT_FIROUND_S()? I can't seem to find even from latest RI-2021.6 tools.
DOC:
image

As far as I know, XT_FIROUND_S and XT_ROUND_S will not exist at the same time.

I just refer to the CANDENCE NNLIB code

**Review by singalsu — APPROVED** [review]
Looks good, thanks for addressing this!

**Review by dbaluta — APPROVED** [review]

**Review by paulstelian97 — APPROVED** [review]

**Pull request merged**
